VideoWeb/VideoWeb/Helpers/ConsultationInvitationTracker.cs

```diff
@@ -0,0 +1,71 @@
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using VideoWeb.Common.Caching;
+using VideoWeb.Common.Models;
+
+namespace VideoWeb.Helpers
+{
+    public interface IConsultationInvitationTracker
+    {
+        Task<Guid> StartTrackingInvitation(Conference conference, string roomLabel, Guid requestedParticipantId);
+        Task<ConsultationInvitation> GetInvitation(Guid invitationId);
+        Task UpdateConsultationResponse(Guid invitationId, Guid participantId, ConsultationAnswer answer);
+        Task<bool> HaveAllParticipantsAccepted(Guid invitationId);
+        Task<bool> HaveAllParticipantsResponded(Guid invitationId);
+    }
+
+    public class ConsultationInvitationTracker : IConsultationInvitationTracker
+    {
+        private readonly IConsultationInvitationCache _consultationInvitationCache;
+
+        public ConsultationInvitationTracker(IConsultationInvitationCache consultationInvitationCache)
+        {
+            _consultationInvitationCache = consultationInvitationCache;
+        }
+
+        public async Task<Guid> StartTrackingInvitation(Conference conference, string roomLabel, Guid requestedParticipantId)
+        {
+            var requestedParticipant = conference.Participants.FirstOrDefault(p => p.Id == requestedParticipantId);
+
+            if (requestedParticipant == null) 
```

Comment:
what would happen if more than one VHO call is incoming for the hearing?

Reply:
It would create an invitation for each call from each unique VHO. This checks the requested participant exists and if it does creates a new invitation.